hudi-common/src/main/java/org/apache/hudi/common/table/timeline/LSMTimeline.java

```diff
@@ -0,0 +1,285 @@
+   */
+  public static HoodieLSMTimelineManifest latestSnapshotManifest(HoodieTableMetaClient metaClient) throws IOException {
+    int latestVersion = latestSnapshotVersion(metaClient);
+    return latestSnapshotManifest(metaClient, latestVersion);
+  }
+
+  /**
+   * Reads the file list from the manifest file for the latest snapshot.
+   */
+  public static HoodieLSMTimelineManifest latestSnapshotManifest(HoodieTableMetaClient metaClient, int latestVersion) {
+    if (latestVersion < 0) {
+      // there is no valid snapshot of the timeline.
+      return HoodieLSMTimelineManifest.EMPTY;
+    }
+    // read and deserialize the valid files.
+    byte[] content = FileIOUtils.readDataFromPath(metaClient.getFs(), getManifestFilePath(metaClient, latestVersion)).get();
+    try {
+      return HoodieLSMTimelineManifest.fromJsonString(new String(content, StandardCharsets.UTF_8), HoodieLSMTimelineManifest.class);
+    } catch (Exception e) {
+      throw new HoodieException("Error deserializing manifest entries", e);
+    }
+  }
+
+  /**
+   * Returns the full manifest file path with given version number.
+   */
+  public static Path getManifestFilePath(HoodieTableMetaClient metaClient, int snapshotVersion) {
+    return new Path(metaClient.getArchivePath(), MANIFEST_FILE_PREFIX + snapshotVersion);
+  }
+
+  /**
+   * Returns the full version file path with given version number.
+   */
+  public static Path getVersionFilePath(HoodieTableMetaClient metaClient) {
+    return new Path(metaClient.getArchivePath(), VERSION_FILE_NAME);
+  }
+
+  /**
+   * List all the parquet manifest files.
+   */
+  public static FileStatus[] listAllManifestFiles(HoodieTableMetaClient metaClient) throws IOException {
+    return metaClient.getFs().listStatus(new Path(metaClient.getArchivePath()), getManifestFilePathFilter());
+  }
+
+  /**
+   * List all the parquet metadata files.
+   */
+  public static FileStatus[] listAllMetaFiles(HoodieTableMetaClient metaClient) throws IOException {
+    return metaClient.getFs().globStatus(
+        new Path(metaClient.getArchivePath() + "/*.parquet"));
+  }
+
+  /**
+   * Parse the snapshot version from the manifest file name.
+   */
+  public static int getManifestVersion(String fileName) {
+    return Integer.parseInt(fileName.split("_")[1]);
+  }
+
+  /**
+   * Parse the layer number from the file name.
+   */
+  public static int getFileLayer(String fileName) {
+    try {
+      Matcher fileMatcher = ARCHIVE_FILE_PATTERN.matcher(fileName);
+      if (fileMatcher.matches()) {
+        return Integer.parseInt(fileMatcher.group(3));
+      }
+    } catch (NumberFormatException e) {
+      // log and ignore any format warnings
+      LOG.warn("error getting file layout for archived file: " + fileName);
+    }
+
+    // return default value in case of any errors
+    return 0;
+  }
+
+  /**
+   * Parse the minimum instant time from the file name.
+   */
+  public static String getMinInstantTime(String fileName) {
```

Comment:
instead of individual parsing methods, can introduce a POJO here, with getters. i.e 

LSMFile class with min, max, level as fields?

Reply:
Did see much gains here, we can extend it in the near future if we have more complex name parsing on the file names.